src/ts/utils/test/block.spec.ts

```diff
@@ -0,0 +1,95 @@
+import { removeSkippedRanges } from "../block";
+
+describe("Block utils", () => {
+  describe("removeSkippedRanges", () => {
+    it("should remove the passed skipped range from the block text", () => {
+      const skipRanges = [{ from: 18, to: 25 }];
+      const block = {
+        id: "id",
+        text: "Example [noted ]text",
+        from: 10,
+        to: 28,
+        skipRanges
+      };
+      const newBlock = removeSkippedRanges(block);
+      expect(newBlock.text).toBe("Example text");
+      expect(newBlock.from).toBe(10);
+      expect(newBlock.to).toBe(22);
+    });
+
+    it("should remove multiple adjacent skipped ranges from the block text", () => {
+      const skipRanges = [
+        { from: 18, to: 25 },
+        { from: 26, to: 32 }
+      ];
+      const block = {
+        id: "id",
+        text: "Example [noted][noted ]text",
+        from: 10,
+        to: 37,
+        skipRanges
+      };
+      const newBlock = removeSkippedRanges(block);
+
+      expect(newBlock.text).toBe("Example text");
+      expect(newBlock.from).toBe(10);
+      expect(newBlock.to).toBe(22);
+    });
+
+    it("should remove multiple non-adjacent skipped ranges from the block text - 1", () => {
+      const skipRanges = [
+        { from: 18, to: 25 },
+        { from: 41, to: 48 }
+      ];
+      const block = {
+        id: "id",
+        text: "Example [noted ]text with more [noted ]text",
+        from: 10,
+        to: 52,
+        skipRanges
+      };
+      const newBlock = removeSkippedRanges(block);
+      expect(newBlock.text).toBe("Example text with more text");
+      expect(newBlock.from).toBe(10);
+      expect(newBlock.to).toBe(37);
+    });
+
+    it("should remove multiple non-adjacent skipped ranges from the block text - 2", () => {
+      const skipRanges = [
+        { from: 40, to: 42 },
+        { from: 44, to: 45 },
+        { from: 47, to: 48 }
+      ];
+      const block = {
+        id: "1602586488022-from:40-to:214",
+        text: "ABCDEFGHIJ",
+        from: 40,
+        to: 47,
+        skipRanges
+      };
+      const newBlock = removeSkippedRanges(block);
+
+      expect(newBlock.text).toBe("DGJ");
+      expect(newBlock.from).toBe(40);
+      expect(newBlock.to).toBe(43);
+    });
+
+    it("should not care about order", () => {
```

Comment:
It might be helpful to have some more awkward cases. For instance if ranges overlap or go beyond the boundaries of the block. Regardless of the behaviour it would be useful to have tests.

Reply:
I've added tests for overlap and going beyond block range boundaries on both sides (thankfully without changing code!) – all good shouts.